src/plan.ts

```diff
@@ -127,6 +127,75 @@ async function planBuildDockerImage(
   })
 }
 
+async function planBuildFileResource(
+  workingDir: string,
+  id: string
+): Promise<BuildPlan[]> {
+  const allCharmcraftFiles = await (
+    await glob.create(path.join(workingDir, '**', 'charmcraft.yaml'))
+  ).glob()
+  const charmcraftFiles = allCharmcraftFiles.filter(
+    file =>
+      !path.normalize(path.relative(workingDir, file)).startsWith('tests/')
+  )
+  return charmcraftFiles.flatMap((charmcraftFile: string) => {
+    const file = path.join(
+      workingDir,
+      path.relative(workingDir, charmcraftFile)
+    )
+    const charmcraft = yaml.load(
+      fs.readFileSync(charmcraftFile, { encoding: 'utf-8' })
+    ) as object
+    const metadataFile = path.join(
+      path.dirname(charmcraftFile),
+      'metadata.yaml'
+    )
+    const metadataExists = fs.existsSync(metadataFile)
+    const metadata = metadataExists
+      ? (yaml.load(
+          fs.readFileSync(metadataFile, { encoding: 'utf-8' })
+        ) as object)
+      : {}
+
+    let charmName: string = ''
+    if ('name' in charmcraft) {
+      charmName = charmcraft['name'] as string
+    } else if ('name' in metadata) {
+      charmName = metadata.name as string
+    } else {
+      throw new Error(`unknown charm name (${workingDir})`)
+    }
+
+    let resources: Map<string, CharmResource> = new Map()
+    if ('resources' in charmcraft) {
+      resources = charmcraft['resources'] as Map<string, CharmResource>
+    }
+    if ('resources' in metadata) {
+      resources = metadata['resources'] as Map<string, CharmResource>
+    }
+
+    return Object.entries(resources).reduce(
+      (acc, [resourceName, resource]: [string, CharmResource]) => {
+        if (resource.type === 'file' && resource.filename) {
+          let parent = path.dirname(file)
+          acc.push({
+            type: 'file',
+            name: resourceName,
+            source_file: resource.filename,
```

Comment:
Maybe we can put the build filename (`build-{resource-name}.sh`) in the `source_file` field and add a new field (maybe called `build_target`?) to include the `resource.filename`. This would better align with other types of plans.

Reply:
see my latest commit and let me know if this is what is on your mind